File: my-folder/0202-happy-number/solution.java
class Solution {
    public boolean isHappy(int n) {
        if (n == 1) {
            return true;
        } else if (n == 4) {
            return false;
        }
        
        int happy = 0;
        while (n > 0) {
            happy += (n % 10) * (n % 10);
            n /= 10;
        }
        
        return isHappy(happy);
    }
}

File: my-folder/0202-happy-number/solution.py
class Solution:
    def isHappy(self, n: int) -> bool:
        if n == 1: return True
        elif n == 4: return False

        happy = 0
        while n > 0:
            happy += (n % 10) * (n % 10)
            n //= 10
        return self.isHappy(happy)
